### PR TITLE
Suggesting articles with titles made of punctuation only

### LINKED ISSUE #763: & symbol in search box still not redirecting to article
From kiwix-tools created by [nijazm](https://github.com/nijazm): kiwix/kiwix-tools#588  

Okay, looks like you were fixing something but unsuccessfully. I just tested yesterday's nightly version of kiwix desktop and kiwix tools on Windows 11. Now just shows fulltext search autocomplete result for & symbol and when I click on it, it says `No results were found for "&"`. In search box it shows containing '&'. The same happens in kiwix serve (web browsers) and kiwix desktop app. Tested with english wikipedia 2021-12. The only difference is that now titles containing & redirect properly (previously they did not), e.g. `Me, Myself & Irene`

### PR DESCRIPTION
**Pull request by veloman-yunkan**
Fixes #763

### CODE REVIEW

**Comment by kelson42** [comment]
@veloman-yunkan Thank you for this imoortant PR. It seems to be 100% on the reader side, which is a good news if confirmed. Have we trsted that it fully fix the original bug report at https://github.com/kiwix/kiwix-tools/issues/588 ?

**Comment by veloman-yunkan** [comment]
> It seems to be 100% on the reader side, which is a good news if confirmed.

@kelson42 I have to disappoint you. Part of the fix is on the writer side: https://github.com/openzim/libzim/pull/765/files#diff-197b97aa6c4de4888c0ee7997d13633629d11fd9999aaec7933dd59bd558c8e3

File: src/suggestion.cpp [hunk 1]
```diff
@@ -139,7 +139,11 @@ Xapian::Query SuggestionDataBase::parseQuery(const std::string& query)
   m_queryParser.set_stemming_strategy(Xapian::QueryParser::STEM_SOME);
   xquery = m_queryParser.parse_query(query, flags);
 
-  if (!query.empty()) {
+  if ( !query.empty() && xquery.get_num_subqueries() == 0 ) {
+    // a non-empty query string produced an empty xapian query which means
+    // that the query string is made solely of punctuation.
```
Comment (mgautierfr) [inline src/suggestion.cpp:144]: "of punctuation or stop words" ?

**Review by mgautierfr — CHANGES_REQUESTED** [review]

File: src/suggestion.cpp (hunk 1 above)
Comment (veloman-yunkan) [reply src/suggestion.cpp:144]: Currently, stop-words are not used for suggestions.

File: src/writer/xapianIndexer.cpp [hunk 2]
```diff
@@ -147,6 +157,11 @@ void XapianIndexer::indexTitle(const std::string& path, const std::string& title
   if (!unaccentedTitle.empty()) {
     std::string anchoredTitle = ANCHOR_TERM + unaccentedTitle;
     indexer.index_text(anchoredTitle, 1);
+    if ( getTermCount(currentDocument) == 1 ) {
+      // only ANCHOR_TERM was added, hence unaccentedTitle is made solely of
+      // non-word characters. Then add entire title as a single term.
+      currentDocument.add_term(unaccentedTitle);
+    }
```
[the remark this answers is not in the document]
Comment (veloman-yunkan) [reply src/writer/xapianIndexer.cpp]: Done

**Pull request merged**